WG_GO_CORE.md

```diff
@@ -0,0 +1,177 @@
+   * There are examples and documentation for ipfs users to create private-content dapps using IPFS (aka handling encryption on the client side)
+* `M (PX)`: IPFS is usable for building wordpress-level decentralized sites
+   * API doesn't suck (and is fast)
+   * We can build an interactive blogging engine with a built-in editor/publisher entirely in a webapp (backed by a go-ipfs instance). And it's *easy* to do this.
+
+### 🤝 IPFS Contributors and Developers
+
+Developers using IPFS can rely on go-ipfs as a platform on which to build their product, app, or ecosystem, and new contributors are well supported.
+
+* `M (P0)`: (all) Go-IPFS is approachable as a new contributor
+   * Go-IPFS internals are well documented
+   * Technical debt is paid off, all "indefinitely-in-progress" refactors are completed.
+   * Cleanup/rethink abstraction layers with 20-20 hindsight.
+   * Re-structure repos into reusable but not fragmented components.
+* `M (P0)`: (all) Go-IPFS is usable without gx
+   * Fix the versions (make them all sub-0)
+   * Add go.mod files to every package
+   * Merge #5435.
+* `M (P0)`: (all) Users can reliably transfer data between any two nodes.
+   * Reliable NAT traversal (AutoNAT, relay, TURN, etc).
+   * Connection manager doesn't kill useful connections.
+   * Scalable content routing
+   * Reliable DHT: unreliable nodes don't join the DHT.
+* `M (P0)`: (dapps) ipfs:// and ipns:// work in web browsers
+   * Base32 CIDs
+   * Base32 IPNS or IPNS uses CIDs
+* `M (P1)`: (all) Go-IPFS is approachable as a user/app developer
+   * User documentation.
+   * Well designed API interface
+   * Well designed API transport
+* `M (P1)`: (Dapps) Fast (< 3s), mutable name resolution (IPNS)
+   * Reliable DHT.
+   * QUIC (for fast connection establishment).
+   * Better protocol negotiation (multistream-2.0)
+   * Delegated Routing
+* `M (P1)`: (Dapps) IPFS Realtime Story is as good as using a centralized service (e.g. Socket.io, Pusher, etc)
+   * The API for doing secure (authenticated) broadcast updates exists 
+* `M (P2)`: Go-ipfs can handle large datasets (>1TiB, >1M nodes)
+   * Scalable content routing (providing)
+   * DagSync (or at least better bitswap)
+   * Reliable, performant datastore
+* `M (P2)`: (all, Dapps) IPFS is pluggable and extensible (9/10 users don't need custom features)
+   * The go-ipfs plugins system is expanded to support new datastores, exchanges, etc
+   * Exchanges support multiple protocols (that diagram from hack week)
+   * Multi-DHT (https://github.com/ipfs/notes/issues/291#issuecomment-414495124)
+   * UnixFS-V2 is implemented
+* `M (P2)`: (machine-learning/package managers) IPFS can handle (and transfer) large (>1M entries) sharded indexes (objects, directories)
+   * Bitswap Improvements (sessions, prediction, etc)
+   * UnixFS-V2 (better directory structure)
+* `M (P3)`: (Dapps) IPFS can locally share data without a shared network

```

Comment:
@mgoelzer @raulk - FYI a bluetooth(like) transport for locally sharing data without a shared network is a requested item in our go-ipfs 2019 roadmap

Reply:
Hey, thanks for pinging me on this.  I'm really interested in the BT transport (and Wifi), but I'm not sure who would work on it.  From past conversations with @whyrusleeping I believe it requires someone who has done low-level Bluetooth development before, and there is no one at PL (afaik) who falls in this category.  We discussed this at the Lisbon Hack Week in May 2018 and I believe @mafintosh (think it was him?) said he knew someone with this experience, but we never followed up on it.  @whyrusleeping also has done some research on this in the past, but lately he's been focusing on FC.

If 2019 is the year that we get serious about a BT transport in libp2p, I think we should put our heads together and (1) create an RFP with acceptance criteria for exactly what we want, and (2) think of who we know who might be able to take this on.

Reply:
Added a crosslink in the libp2p roadmap: https://docs.google.com/document/d/1Rd4yNw1TNQBvfRrKeEMSTseb6fvPzS-C--obOn0nul8/edit?disco=AAAACYoDclo We need to get better at tracking these crosslinks in a visual manner.